tests/apple-exporter/components/test-class-image.php

```diff
@@ -30,6 +30,28 @@ public function filter_apple_news_image_json( $json ) {
 		return $json;
 	}
 
+	/**
+	 * Test component type ('image' or 'photo') postmeta selection.
+	 */
+	public function test_component_type() {
+		// Create test post.
+		$post_id = self::factory()->post->create(
+			[
+				// First img becomes cover, second img is output as standard json component.
+				'post_content' => '<img src="https://placeimg.com/640/480/any" alt="Example Cover" /><img src="https://placeimg.com/640/480/any" alt="Example Image" />',
+			]
+		);
+
+		// Toggle component type meta.
+		update_post_meta( $post_id, 'apple_news_use_image_component', true );
+
+		$json = $this->get_json_for_post( $post_id );
+		// Test cover image role.
+		$this->assertEquals( 'image', $json['components'][0]['components'][0]['role'] );
+		// Test second image role.
+		$this->assertEquals( 'image', $json['components'][1]['components'][3]['role'] );
```

Comment:
You should test that this is `photo` and then update the postmeta and then get the JSON for the post again, and then test it has become `image`